Add PostAction for dotnet.config

Youssef1313 opened the pull request:
Related to #50054

Youssef1313 committed:
Add PostAction for dotnet.config

Youssef1313 committed:
Use stream

Youssef1313 committed:
Fix

Youssef1313 committed:
Fix tests

Youssef1313 committed:
Fix tests

Youssef1313 committed:
Address Copilot comments

Youssef1313 commented:
@dotnet/templating-engine-maintainers @mariam-abdulla @nohwnd Please review

Youssef1313 committed:
Update src/Cli/Microsoft.TemplateEngine.Cli/PostActionProcessors/CreateOrUpdateDotnetConfigPostActionProcessor.cs

Youssef1313 committed:
Reduce noise

Youssef1313 commented:
@joeloff Yes I'll backport after merge. And for next PRs I'll target release/10.0.1xx right away

Youssef1313 committed:
Update src/Cli/Microsoft.TemplateEngine.Cli/LocalizableStrings.resx

Co-authored-by: Jacques Eloff <joeloff@users.noreply.github.com>

Youssef1313 committed:
Update xlf

nohwnd commented:
Suggestion: 

Looking at the post action that adds project to solution, there are some more things that will need to happen (probably), adding the guid of the post action to templating schema.

And the tests that actually use the action would also be nice to have here:

https://grep.app/search?f.repo.pattern=dotnet%2F&q=D396686C-DE0E-4DE6-906D-291CD29FC5DE

Maybe if this post action moved to the location of https://github.com/dotnet/sdk/blob/main/src/Cli/dotnet/Commands/New/PostActions/DotnetSlnPostActionProcessor.cs#L24

it would be nice to re-use that infrastructure and do it the same.

Youssef1313 committed:
Add more tests

Youssef1313 commented:
@nohwnd I addressed your feedback on adding more tests, and opened a doc update on dotnet/templating dotnet/templating#9274. For reusing logic from DotnetSlnPostActionPreprocessor, my logic here is quite different, which is more suitable for 'dotnet.config' IMO.

Ping @dotnet/templating-engine-maintainers @marcpopMSFT any concerns merging this?

Youssef1313 committed:
Refactor tests to be not specific to MTP

nohwnd committed:
Fix template tests
